extensions/reactive-routes/deployment/src/test/java/io/quarkus/vertx/web/cors/CORSHandlerTestCase.java

```diff
@@ -31,25 +31,23 @@ public void corsPreflightTestServlet() {
                 .options("/test").then()
                 .statusCode(200)
                 .header("Access-Control-Allow-Origin", origin)
-                .header("Access-Control-Allow-Methods", methods)
+                .header("Access-Control-Allow-Methods", "GET,OPTIONS,POST")
                 .header("Access-Control-Allow-Headers", headers);
     }
 
     @Test
     @DisplayName("Handles a direct CORS request correctly")
     public void corsNoPreflightTestServlet() {
         String origin = "http://custom.origin.quarkus";
-        String methods = "GET,POST";
+        String methods = "POST";
         String headers = "X-Custom";
         given().header("Origin", origin)
-                .header("Access-Control-Request-Method", methods)
```

Comment:
Note that these two headers are only valid for preflight (i.e. OPTIONS) requests. They should be ignored otherwise as they have no defined meaning outside an options request.

Reply:
@stuartwdouglas `Origin` can be sent with simple requests, we've had the related issues